apps/desktop-wallet/src/modals/AddressDetailsModal.tsx

```diff
@@ -32,30 +32,33 @@ import HashEllipsed from '@/components/HashEllipsed'
 import QRCode from '@/components/QRCode'
 import TransactionList from '@/components/TransactionList'
 import AssetsTabs from '@/features/assetsLists/AssetsTabs'
-import { useAppSelector } from '@/hooks/redux'
-import CSVExportModal from '@/modals/CSVExportModal'
-import ModalPortal from '@/modals/ModalPortal'
+import { closeModal, openModal } from '@/features/modals/modalActions'
+import { ModalBaseProp } from '@/features/modals/modalTypes'
+import { useAppDispatch, useAppSelector } from '@/hooks/redux'
 import SideModal from '@/modals/SideModal'
 import AmountsOverviewPanel from '@/pages/UnlockedWallet/OverviewPage/AmountsOverviewPanel'
 import { selectAddressByHash } from '@/storage/addresses/addressesSelectors'
 import { openInWebBrowser } from '@/utils/misc'
 
-interface AddressDetailsModalProps {
+export interface AddressDetailsModalProps {
   addressHash: AddressHash
-  onClose: () => void
 }
 
-const AddressDetailsModal = ({ addressHash, onClose }: AddressDetailsModalProps) => {
+const AddressDetailsModal = memo(({ id, addressHash }: ModalBaseProp & AddressDetailsModalProps) => {
   const { t } = useTranslation()
   const theme = useTheme()
   const address = useAppSelector((s) => selectAddressByHash(s, addressHash))
   const explorerUrl = useAppSelector((s) => s.network.settings.explorerUrl)
+  const dispatch = useAppDispatch()
 
-  const [isCSVExportModalOpen, setIsCSVExportModalOpen] = useState(false)
   const [showChart, setShowChart] = useState(false)
 
   if (!address) return null
 
+  const onClose = () => dispatch(closeModal({ id }))
```

Comment:
Once all modals have been refactored, there will be no need to declare this function in every modal. It can be moved to the `ModalContainer`. But until then, to keep the code functional, we need to provide an `onClose` function.

Reply:
cf. my idea to create a `useModal` hook which could return a `closeModal` function.

Reply:
I summed up my findings in the other comment: https://github.com/alephium/alephium-frontend/pull/810#discussion_r1740566632